js/src/timebomb.ts

```diff
@@ -49,7 +63,28 @@ export function failAfter(time: Date, options: Partial<Config> = {}) {
   }
 }
 
-export function slowAfter(
+export async function slowAfter(
```

Comment:
NB: I'd prefer if we managed to figure out a way to share the logic between slowAfter and slowAfterSync-- maybe a generic higher-order-function that returns the result of wait?

```
function buildSlowAfter<T>(waitFn: ((number) => A)){
  return (time, delay, options) => {
    ...
  }
}

const slowAfter = buildSlowAfter(wait);
const slowAfterSync = buildSlowAfter(waitSync);
```

Reply:
If this lib (like right now) doesn't need to do anything after the wait, we could wrap the loop in a promise (since the executer gets executed in the same callstack if I'm not mistaken).

This would make a higher order function easier to implement.

Reply:
Sounds good! Do whatever you think is cleanest -- i just proposed what I thought might work cleanly.